### PR TITLE
Reject unsupported signal events

### LINKED ISSUE #12008: Reject processes with unsupported signal events
**Description**

With #11194 completed, the upcoming minor release `8.2` will support top-level signal start events. However, the other signal events (See #10777) are not yet supported.

Deployed processes with unsupported signal events must be rejected to avoid unexpected behavior:
- Event subprocess signal start events (interrupting and non-interrupting)
- All intermediate signal events (catch and throw)
- Signal end events

### PR DESCRIPTION
**Pull request by korthout**
## Description

Rejects deployments of processes containing any of the following (currently) unsupported signal events:
- signal end event
- signal throw event
- signal boundary event
- signal intermediate catch event
- signal event subprocess

This is necessary because these events are currently unsupported and would result in unexpected behavior when executed.

## Related issues

closes #12008 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [x] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by korthout** [comment]
@remcowesterhoud @berkaycanbc who-ever has time first, please have a look :)

File: engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/validation/UnsupportedSignalEventSubprocessValidator.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    final Collection<StartEvent> startEvents = subProcess.getChildElementsByType(StartEvent.class);
+    startEvents.stream()
+        .map(CatchEvent::getEventDefinitions)
+        .filter(e -> e.stream().anyMatch(SignalEventDefinition.class::isInstance))
+        .forEach(
+            signalSubProcess ->
+                validationResultCollector.addError(
+                    0, "Elements of type signal event subprocess are currently not supported"));
```
Comment (remcowesterhoud) [inline engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/validation/UnsupportedSignalEventSubprocessValidator.java:40]:
```suggestion
                    0, "Elements of type signal start event in an event subprocess are currently not supported"));
```

🙃 Signal event subprocess is not really an element type. This also requires a change in the tests

**Review by remcowesterhoud — APPROVED** [review]
Thanks @korthout 🚀 I have one minor suggestion, please have a look!

**Comment by korthout** [comment]
bors merge

**Pull request merged**